e2etest/bgp_test.go

```diff
@@ -465,18 +474,150 @@ var _ = ginkgo.Describe("BGP", func() {
+				framework.ExpectNoError(err)
+
+				for _, c := range frrContainers {
+					validateFRRPeeredWithNodes(cs, c)
+				}
+
+				ginkgo.By(fmt.Sprintf("configure service number %d", i+1))
+				svc, _ := createServiceWithBackend(cs, f.Namespace.Name, fmt.Sprintf("svc%d", i+1), func(svc *corev1.Service) {
+					svc.Spec.ExternalTrafficPolicy = corev1.ServiceExternalTrafficPolicyTypeCluster
+					svc.Annotations = map[string]string{"metallb.universe.tf/address-pool": fmt.Sprintf("test-addresspool%d", i+1)}
+				})
+
+				defer func() {
+					err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+					framework.ExpectNoError(err)
+				}()
+
+				ginkgo.By("validate LoadBalancer IP is in the AddressPool range")
+				ingressIP := e2eservice.GetIngressPoint(
+					&svc.Status.LoadBalancer.Ingress[0])
+				err = validateIPInRange([]addressPool{pool}, ingressIP)
+				framework.ExpectNoError(err)
+
+				services = append(services, svc)
+				servicesIngressIP = append(servicesIngressIP, ingressIP)
+
+				for j := 0; j <= i; j++ {
+					ginkgo.By(fmt.Sprintf("validate service %d IP didn't change", j+1))
+					ip := e2eservice.GetIngressPoint(&services[j].Status.LoadBalancer.Ingress[0])
+					framework.ExpectEqual(ip, servicesIngressIP[j])
+
+					ginkgo.By(fmt.Sprintf("checking connectivity of service %d to its external VIP", j+1))
+					for _, c := range frrContainers {
+						validateService(cs, services[j], allNodes.Items, c, ipFamily)
+					}
+				}
+			}
+		},
+			table.Entry("IPV4", "192.168.10.0/24", "ipv4"),
+			table.Entry("IPV6", "fc00:f853:0ccd:e799::/116", "ipv6"))
+
+		table.DescribeTable("configure peers one by one and validate FRR paired with nodes", func(ipFamily string) {
+			var peers []peer
+
+			for i, c := range frrContainers {
+				ginkgo.By("configure peer")
+				c.RouterConfig.IPFamily = ipFamily
+				c.NeighborConfig.IPFamily = ipFamily
+				address := c.Ipv4
+				if ipFamily == "ipv6" {
+					address = c.Ipv6
+				}
+				peers = append(peers, peer{
+					Addr:     address,
+					ASN:      c.RouterConfig.ASN,
+					MyASN:    c.NeighborConfig.ASN,
+					Port:     c.RouterConfig.BGPPort,
+					RouterID: c.RouterConfig.RouterID,
+					Password: c.RouterConfig.Password,
+					HoldTime: c.RouterConfig.HoldTime,
+				})
+
+				configData := configFile{
+					Peers: peers,
+				}
+				err := updateConfigMap(cs, configData)
+				framework.ExpectNoError(err)
+
+				pairExternalFRRWithNodes(cs, c)
+
+				for j := 0; j <= i; j++ {
+					validateFRRPeeredWithNodes(cs, frrContainers[j])
+				}
+			}
+		},
+			table.Entry("IPV4", "ipv4"),
+			table.Entry("IPV6", "ipv6"))
+	})
 })
 
-func createServiceWithBackend(cs clientset.Interface, namespace string, policy corev1.ServiceExternalTrafficPolicyType) (*corev1.Service, *e2eservice.TestJig) {
+func createServiceWithBackend(cs clientset.Interface, namespace string, jigName string, config func(*corev1.Service)) (*corev1.Service, *e2eservice.TestJig) {
```

Comment:
Sorry I wasn't clear, my suggestion about the callback(s) (as a variadic args) was to be able to override the service without adding an extra parameter.
I see here that for the name we need an extra param anyway, so there's no point in moving to the function.

Reply:
i think it's more clean to have the function this way so we can set the service parameters. what do you think?

Reply:
I think it will be clear if we'll need to add more parameters (which might be the case when https://github.com/metallb/metallb/pull/1065 lands. Adding this to change always the single parameter is a bit of an overcomplication imo.

In anycase, since we'll need to do that, I think the calling site would benefit from two global `config func(*corev1.Service)`, one as SetClusterTrafficPolicy and the other as SetLocalTrafficPolicy so you can use those as parameters for this function.